src/nvim/search.c

```diff
@@ -1268,13 +1268,13 @@ int do_search(oparg_T *oap, int dirc, int search_delim, char_u *pat, long count,
           while (*r == ' ') {
             r++;
           }
-          size_t pat_len = msgbuf + STRLEN(msgbuf) - r;
+          size_t pat_len = (size_t)(msgbuf - r) + STRLEN(msgbuf);
```

Comment:
???

Reply:
```suggestion
          size_t pat_len = STRLEN(msgbuf) - (size_t)(r - msgbuf);
```

Reply:
I'm casting the `ptrdiff_t` to `size_t`. I'll revert it and just cast the entire thing instead.